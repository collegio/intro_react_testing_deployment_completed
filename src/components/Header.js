import React from 'react';
import { Link } from 'react-router-dom';

const Header = (props) => {
    return (
        <div className="header">
            <div className="container">
                <div className="nav-links">
                    <Link to="/">Home</Link>&nbsp;&nbsp;|&nbsp;&nbsp;
                    <Link to="/create">Add Activity!</Link>
                </div>
                <h1>{props.children}</h1>
                <h2>{props.subtitle}</h2>
                
            </div>
        </div>
    );
}
Header.defaultProps = {
    subtitle: "Get active after work!"
};

export default Header;