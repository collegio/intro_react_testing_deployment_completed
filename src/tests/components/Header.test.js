import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

test ('should render the header', () => {

    const wrapper = shallow(<Header>Test Header</Header>);

    expect(wrapper.find('h1').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
});