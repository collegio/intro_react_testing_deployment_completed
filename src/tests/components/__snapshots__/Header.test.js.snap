// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the header 1`] = `
<div
  className="header"
>
  <div
    className="container"
  >
    <div
      className="nav-links"
    >
      <[object Object]
        to="/"
      >
        Home
      </[object Object]>
        |  
      <[object Object]
        to="/create"
      >
        Add Activity!
      </[object Object]>
    </div>
    <h1>
      Test Header
    </h1>
    <h2>
      Get active after work!
    </h2>
  </div>
</div>
`;
